Add tags field to Metstore Service for TagsR2401

SamanthaMathews opened the pull request:
Add tags field to service resource to allow setting tags on service resources at creation time.
Part of b/365716008

The contents of this code are entirely owned by Google LLC in accordance with the agreement between Google LLC and the third party submitting this code into Google's open source repository

Release Note Template for Downstream PRs 
```
metastore: added `tags` field to `metastore_service` to allow setting tags for services at creation time
```

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report

Your PR generated some diffs in downstreams - here they are.

`google` provider: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-11611-old..auto-pr-11611) ( 4 files changed, 146 insertions(+), 12 deletions(-))
`google-beta` provider: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-11611-old..auto-pr-11611) ( 4 files changed, 147 insertions(+), 13 deletions(-))
`terraform-google-conversion`: [Diff](https://github.com/modular-magician/terraform-google-conversion/compare/auto-pr-11611-old..auto-pr-11611) ( 1 file changed, 17 insertions(+))



## Missing test report
Your PR includes resource fields which are not covered by any test.

Resource: `google_dataproc_metastore_service` (26 total tests)
Please add an acceptance test which includes these fields. The test should include the following:

```hcl
resource "google_dataproc_metastore_service" "primary" {
  tags = # value needed
}


```

modular-magician commented:
#### Tests analytics
Total tests: 0
Passed tests: 0
Skipped tests: 0
Affected tests: 0

<details>
<summary>Click here to see the affected service packages</summary>
<blockquote>

<ul>
<li>dataprocmetastore</li>

</ul>

</blockquote>
</details>
#### Non-exercised tests

Tests were added that are skipped in VCR:
- TestAccMetastoreService_tags
$\textcolor{red}{\textsf{Errors occurred during REPLAYING mode. Please fix them to complete your PR.}}$

View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-11611/artifacts/228fa86e-992e-41d8-a22a-f3c967770229/build-log/replaying_test.log)

NickElliot reviewed (changes requested):
pending googler erroneous cla trouble shooting see: https://github.com/GoogleCloudPlatform/magic-modules/pull/11555#pullrequestreview-2281147930

abd-goog commented on mmv1/products/metastore/Service.yaml:
```diff
@@ -515,3 +515,12 @@ properties:
+    description: |
+      A map of resource manager tags.
+      Resource manager tag keys and values have the same definition as resource manager tags.
+      Keys must be in the format tagKeys/{tag_key_id}, and values are in the format tagValues/456.
```
nit: can be tagValues/{tag_value_id} instead of tagValues/456
here and other instances in the PR.

SamanthaMathews replied:
Done

abd-goog commented on mmv1/third_party/terraform/services/dataprocmetastore/resource_dataproc_metastore_service_test.go.erb:
```diff
@@ -107,6 +107,34 @@ func TestAccDataprocMetastoreService_PrivateServiceConnect(t *testing.T) {
 }
 
+func TestAccMetastoreService_tags(t *testing.T) {
+	t.Skip()
```
why skip?

abd-goog commented on mmv1/third_party/terraform/services/dataprocmetastore/resource_dataproc_metastore_service_test.go.erb:
```diff
@@ -107,6 +107,34 @@ func TestAccDataprocMetastoreService_PrivateServiceConnect(t *testing.T) {
+				ResourceName:            "google_dataproc_metastore_service.default",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"tags"},
```
This probably fails right now as the field service_id, location, labels and terraform_labels are also not read back. So they should also be included in this list for it to pass. Please ensure the test passes.

SamanthaMathews replied:
Sure

abd-goog commented on mmv1/third_party/terraform/website/docs/d/dataproc_metastore_service.markdown:
````diff
@@ -17,6 +18,16 @@ data "google_dataproc_metastore_service" "foo" {
 }
 ```
 
+To create an service with a tag
````
nit : a service

SamanthaMathews replied:
done

abd-goog commented on mmv1/third_party/terraform/website/docs/d/dataproc_metastore_service.markdown:
```diff
@@ -28,6 +39,7 @@ The following arguments are supported:
 
 * `project` - (Optional) The project in which the resource belongs. If it
     is not provided, the provider project is used.
+* `tags` - (Optional) A map of resource manager tags. Resource manager tag keys and values have the same definition as resource manager tags. Keys must be in the format tagKeys/{tag_key_id}, and values are in the format tagValues/456. The field is ignored when empty. The field is immutable and causes resource replacement when mutated.
```
nit: tagValues/{tag_value_id}

SamanthaMathews replied:
done

c2thorn commented:
Discussing with @SamanthaMathews offline about this PR and the go rewrite migration

NickElliot committed:
Convert from Ruby to Go

Delete mmv1/third_party/terraform/services/dataprocmetastore/resource_dataproc_metastore_service_test.go.erb
Update dataproc_metastore_service.markdown
commit terraform-metastore-service-tags
commit terraform-metastore-service-tags
commit terraform-metastore-service-tags
commit terraform-metastore-service-tags
commit terraform-metastore-service-tags
commit terraform-metastore-service-tags

NickElliot committed:
manually resolving

NickElliot commented:
Converted, backed up at https://github.com/NickElliot/magic-modules/tree/terraform-metastore-service-tags-11611-backup

SamanthaMathews committed:
Merge branch 'GoogleCloudPlatform:main' into terraform-metastore-service-tags
